src/test/java/part1/exercise/StreamsExercise2.java

```diff
@@ -121,4 +156,78 @@ public void greatestExperiencePerEmployer() {
         );
     }
 
+    private static Stream<StreamsExercise2.PersonEmployerPair> employeeToPairs(Employee employee) {
+        return employee.getJobHistory().stream()
+                .map(JobHistoryEntry::getEmployer)
+                .map(e -> new StreamsExercise2.PersonEmployerPair(employee.getPerson(), e));
+    }
+
+    private static Stream<StreamsExercise2.PersonEmployerPair> firstEmployeeToPairs(Employee employee) {
+        return employee.getJobHistory().stream()
+                .map(JobHistoryEntry::getEmployer)
+                .limit(1)
+                .map(e -> new StreamsExercise2.PersonEmployerPair(employee.getPerson(), e));
+    }
+
+    private Map<String, Person> getGreatestExperiencePerEmployer(List<Employee> employees) {
+        final Stream<StreamsExercise2.PersonEmployerDuration> personEmployerDurationStream = employees.stream()
+                .flatMap(
+                        e -> e.getJobHistory()
+                                .stream()
+                                .map(j -> new StreamsExercise2.PersonEmployerDuration(e.getPerson(), j.getEmployer(), j.getDuration())));
+
+        return personEmployerDurationStream
+                .collect(groupingBy(
+                        StreamsExercise2.PersonEmployerDuration::getEmployer,
+                        collectingAndThen(
+                                maxBy(comparing(StreamsExercise2.PersonEmployerDuration::getDuration)), p -> p.get().getPerson())));
+    }
+
+    @BeforeClass
+    public static void generateEmployeeList() {
+        employees = getEmployees();
+    }
+
+    @Test
+    public void employersStuffLists() {
+        System.out.println("Employers list stuff:");
+        Map<String, List<Person>> employersStuffLists = employees.stream()
+                .flatMap(StreamsExercise2::employeeToPairs)
+                .collect(Collectors.groupingBy(
+                        StreamsExercise2.PersonEmployerPair::getEmployer,
+                        mapping(StreamsExercise2.PersonEmployerPair::getPerson, toList())));
+
+        employersStuffLists.keySet()

```

Comment:
As you like. Note that you can check not only equality, but also some other properties using `allMatch`, `nonMatch` etc.